### PR TITLE
Edit post: Convert tools more menu group into internal plugin

### PR DESCRIPTION
**Pull request by gziolo**
## Description
This PR removes internal filters included in More Menu groups. They were never publicly promoted so I don't expect them to cause any issues at all. If you still think that we should include a deprecation message, I'm fine to add it. I don't expect any plugin to use it because it would require very careful study of the code. It was used only as an exploration for the `Tools` group and the copy button.

This PR does the following:
- removes all the filters which were meant to make More Menu extensible - this caused lots of confusion recently and raised some question
- adds new internal More Menu group slot for `Tools` 
- adds internal `registerPlugin` call to dogfood Edit Post page with Plugins API

This change will allow use to continue refactoring towards having Edit Post more plugins oriented architecture and a good set of examples to follow by plugins.

## How has this been tested?
There should be no visual change. All More Menu items should look and work as before.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -2421,7 +2421,8 @@
 			"version": "file:packages/shortcode",
 			"requires": {
 				"@babel/runtime": "^7.0.0",
-				"lodash": "^4.17.10"
+				"lodash": "^4.17.10",
+				"memize": "^1.0.5"
```
Comment (gziolo) [inline package-lock.json]: This pops up in every PR I open, we should fix it.

File: packages/components/src/menu-group/index.js [hunk 2]
```diff
@@ -18,14 +17,10 @@ import { NavigableMenu } from '../navigable-container';
...
 	className = '',
-	filterName,
 	instanceId,
 	label,
 } ) {
-	const childrenArray = Children.toArray( children );
-	const menuItems = filterName ?
-		applyFilters( filterName, childrenArray ) :
-		childrenArray;
+	const menuItems = Children.toArray( children );
```
Comment (mcsf) [inline packages/components/src/menu-group/index.js]: Can we now skip `toArray` and go straight for:

```js
if ( ! Children.count( children ) ) return null;

return (
  …
  { children }
```

File: edit-post/plugins/index.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+/**
+ * Internal dependencies
+ */
+import CopyContentMenuItem from './copy-content-menu-item';
+import KeyboardShortcutsHelpMenuItem from './keyboard-shortcuts-help-menu-item';
+import PublishSidebarToggleMenuItem from './publish-sidebar-toggle-menu-item';
+import TipsToggleMenuItem from './tips-toggle-menu-item';
+import ToolsMoreMenuGroup from '../components/header/tools-more-menu-group';
+
+registerPlugin( 'edit-post', {
```
Comment (mcsf) [inline edit-post/plugins/index.js]: 😎

**Review by mcsf — APPROVED** [review]
Nice one!

File: packages/components/src/menu-group/index.js (hunk 2 above)
Comment (gziolo) [reply packages/components/src/menu-group/index.js]: Right, we probably can refactor further 👍

**Comment by gziolo** [comment]
@fastlinemedia - I opened #11570 to track your report 👍

**Comment by fastlinemedia** [comment]
Thanks!

**Pull request merged**